script.js

```diff
@@ -21,102 +21,170 @@ items.item3.element.id = '3';
                     break;
                 }
-            }
-                
+                else {
+                    playerIsCarrying = null; //drop it
+                    items[i].onBelt = false;
+                    //how to distinguish if its on the belt
+                    if (items[i].y <= 70 && items[i].y >= 0) {
+                        items[i].onBelt = true;
+                    }
+                    else {
+                        //make it drop
+                        var posY = items[i].y;
+                         var posX = items[i].x;
+
+
+                        clearInterval(id[i]);
+                        id[i] = setInterval(frame, 10);
+                        function frame() {
+
+                            var itemIsOnScreen = posY < 878;
+                            for (bin in containers){
+                            var curBin = containers[bin];
+                            // this only checks bin1 right now, but should check more
+                            var itemIsAboveBin = posY < curBin.y;
+                            var itemIsInsideLeftWallOfBin = posX >= curBin.x;
+                            console.log(curBin);
+                            var itemIsInsideRightWallOfBin = ((posX+ items[i].element.offsetWidth) <= (curBin.x + curBin.element.offsetWidth));
+                           
+                            
+                            if (!itemIsOnScreen || (!itemIsAboveBin && itemIsInsideLeftWallOfBin && itemIsInsideRightWallOfBin) ) {
+                               if (!itemIsOnScreen){console.log('item is not on the screen');}
+                               else{
+                                //snap in place
+                                posY += curBin.element.offsetHeight-items[i].element.offsetHeight - 10;
+                                items[i].element.style.top = posY + 'px';
+                                console.log(`${i} is inside ${bin}`);
+                                console.log(itemIsAboveBin);
+                                console.log(itemIsInsideLeftWallOfBin);
+                                console.log(itemIsInsideRightWallOfBin);
+                               }
+                                clearInterval(id[i]);
+                            } 
+                            else {
+                                console.log(`${i} is not inside ${bin}`);
+                                posY++;
+                                items[i].y = posY;
+                                items[i].element.style.top  = posY + 'px';
+                            }
+                        }
+
+                        }
+                    }
                 }
-               
             }
+
+
+
         }
-    
+    }
+}
+
+
+
 
-// function dropItem(){
-//     if (playKeys["Space"] == false){
-//         playerIsCarrying = null;
-//     }
-// }
+
+
+
+
+
+var id = { item1: null, item2: null, item3: null, item4: null };
```

Comment:
I think this variable is handling the `setInterval` callbacks for the falling items, but the name doesn't make that super clear - maybe rename this to make it a bit more clear?